ansible/stage2.yml

```diff
@@ -1,13 +1,32 @@
 ---
-- name: Execute getbestchainlock
+
+#We need to check first to ensure mixing is not enabled from a previous run, if it tries to do it again it will fail (thus not idempotent)
+- name: Check mixing status and start mixing if not enabled
   hosts: wallet_nodes
+  become: true
   tasks:
-    - name: Run dash-cli getbestchainlock
-      ansible.builtin.command: dash-cli getbestchainlock
-      register: chainlock_output
-      ignore_errors: true
-
-    - name: Check for chainlock
-      ansible.builtin.fail:
-        msg: "Chainlock failure: no chainlock found."
-      when: '"error" in chainlock_output.stderr'
+
+    - name: Create CoinJoin wallets
+      ansible.builtin.command:
+        cmd: dash-cli createwallet coinjoin-wallet-{{ item | string }} false false "" false true
+        creates: "{{ dashd_home }}/.dashcore/{{ dash_network_name if dash_network == 'devnet' else 'testnet3' }}/wallets/coinjoin-wallet-{{ item | string }}/wallet.dat"
+      loop: "{{ range(0, coinjoin_wallet_count | int, 1) }}"
+
+    - name: Restart dash core
+      community.docker.docker_compose:
+        project_src: '{{ dashd_home }}/dashcore/'
+        state: present
+        restarted: true
+        pull: true
+        timeout: 30
+    
+    - name: Check we're up after restart
+      ansible.builtin.command: dash-cli -rpcwallet=coinjoin-wallet-0 getcoinjoininfo
+      register: coinjoin_output
+      retries: 3
+      delay: 10
+      until: coinjoin_output is not failed
+
+    - name: Include mixing role if coinjoin is not enabled
```

Comment:
Task name implies a condition, but there is no condition here. `mixing` role will run every time, right?